geomagio/JSON/JSONWriter.py

```diff
@@ -0,0 +1,212 @@
+        self.dictionary = dict
+
+    def _format_times(self, timeseries, channels):
+        """format times for json file and update dictionary
+
+        Parameters
+        ----------
+        timeseries: obspy.core.stream
+            timeseries object with data to be written
+        channels: array_like
+            channels to be reported.
+        """
+        times = []
+        traces = [timeseries.select(channel=c)[0] for c in channels]
+        starttime = float(traces[0].stats.starttime)
+        delta = traces[0].stats.delta
+        for i in range(len(traces[0].data)):
+            times.append(self._format_time_string(
+                datetime.utcfromtimestamp(starttime + i * delta)))
+        self.dictionary['times'] = times
+
+    def _format_time_string(self, time):
+        """format one time.
+
+        Parameters
+        ----------
+        time : datetime
+            timestamp for values
+
+        Returns
+        -------
+        unicode
+            formatted time.
+        """
+        tt = time.timetuple()
+        return '{0.tm_year:0>4d}-{0.tm_mon:0>2d}-{0.tm_mday:0>2d}T' \
+                '{0.tm_hour:0>2d}:{0.tm_min:0>2d}:{0.tm_sec:0>2d}.{1:0>3d}Z'.format(
+                tt, int(time.microsecond / 1000))
+
+    def _format_data(self, timeseries, channels, stats):
+        """Format all data lines.
+
+        Parameters
+        ----------
+        timeseries : obspy.core.Stream
+            stream containing traces with channel listed in channels
+        channels : sequence
+            list and order of channel values to output.
+        stats: obspy.core.trace.stats
+            holds the observatory metadata
+        """
+        self.dictionary['values'] = []
+        if timeseries.select(channel='D'):
+            d = timeseries.select(channel='D')
+            d[0].data = ChannelConverter.get_minutes_from_radians(d[0].data)
+        values = []
+        self.edge = EdgeFactory()
+        for c in channels:
+            value_dict = OrderedDict()
+            trace = timeseries.select(channel=c)[0]
+            value_dict['id'] = c
+            value_dict['metadata'] = OrderedDict()
+            value_dict['metadata']['element'] = c
+            if 'network' in stats:
+                value_dict['metadata']['network'] = stats.network
+            value_dict['metadata']['station'] = stats.station
+            edge_channel = self.edge._get_edge_channel(stats.station,
+                                                        c,
+                                                        stats.data_type,
+                                                        stats.data_interval)
+            value_dict['metadata']['channel'] = edge_channel
+            if 'location' in stats:
+                value_dict['metadata']['location'] = stats.location
+            # TODO: Add flag metadata
+            values += [value_dict]
+            data = []
+            for i in range(len(trace.data)):
```

Comment:
data values should be numeric/None.  It's also much more efficient to use numpy methods, instead of looping over the entire array in python:
```
data = np.copy(trace.data)
data[np.isnan(data)] = None
```